### DISCUSSION TITLE
C++: How to filter completion suggestions based on item kind

### DISCUSSION THREAD

**Discussion by jeroen-dhollander** [neoclide/coc.nvim#4692] [Q&A] [answered]
Hey experts,

 When I trigger completion after typing `this->` Coc only suggests the class' members and methods.

 I was wondering if there is a way to trigger this behavior without typing `this->`, for example by filtering suggestion results based on the completion item kind.

 I looked through `coc-suggest-settings` but I couldn't find a way to filter suggestion results.

Is this possible?

Thanks!

Jeroen

**Comment by fannheyward** [comment, marked as answer]
I don't think there's a way to do this, coc.nvim requests to language server, (here is clangd?), with current document and current position, the LS returns available completions. There's no way to request by special item kind.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_completion

**Comment by jeroen-dhollander** [comment]
Thanks for the link to the spec, that's very useful.

This spec talks about a `CompletionItemKind` capability that the client can send to the server; Is there maybe a way to use this to filter responses? Or can this only be changed on server restart?

**Reply by chemzqm** [reply]
> This spec talks about a CompletionItemKind capability 

It should be the supported CompletionItemKind of the client, not suitable for filter.
